From f7ebf2c4d4301ce9aef576b184954af502b186ed Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Sun, 12 Feb 2023 16:11:34 +0800
Subject: [PATCH 030/205] va: jpegdecoder: Do not check SOS state when parsing
 DRI marker.

According to spec, the JPEG_MARKER_DRI(Restart interval definition)
marker can come before the SOS marker. So we should not check the SOS
state when parsing the DRI marker.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/3943>
---
 sys/va/gstjpegdecoder.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sys/va/gstjpegdecoder.c b/sys/va/gstjpegdecoder.c
index ce2c67dba..2448c3cb5 100644
--- a/sys/va/gstjpegdecoder.c
+++ b/sys/va/gstjpegdecoder.c
@@ -481,9 +481,10 @@ gst_jpeg_decoder_handle_frame (GstVideoDecoder * decoder,
         break;
 
       case GST_JPEG_MARKER_DRI:
-        if (!(valid_state (priv->state, GST_JPEG_DECODER_STATE_GOT_SOS)
-                && decode_restart_interval (self, &seg)))
+        if (!decode_restart_interval (self, &seg)) {
+          GST_WARNING_OBJECT (self, "Fail to decode restart interval");
           goto unmap_and_error;
+        }
         break;
       case GST_JPEG_MARKER_DNL:
         break;
-- 
2.40.1

